cohort-engine/src/main/java/com/ibm/cohort/engine/measure/LibraryHelper.java

```diff
@@ -5,27 +5,171 @@
+	 *                                FHIR library resource that contains the
+	 *                                CQL/ELM content
+	 * @param loadedIds               Set of library IDs that have already been loaded by the tree walk. This is used for cycle detection.
+	 * @return List of executable logic libraries
+	 */
+	protected static List<org.cqframework.cql.elm.execution.Library> loadLibraries(org.hl7.fhir.r4.model.Library library,
+			org.opencds.cqf.cql.engine.execution.LibraryLoader libraryLoader,
+			LibraryResolutionProvider<org.hl7.fhir.r4.model.Library> libraryResourceProvider, Set<String> loadedIds) {
+
+		List<org.cqframework.cql.elm.execution.Library> libraries = new ArrayList<>();
+		
+		if( ! loadedIds.contains(library.getId()) ) {
+
+			loadedIds.add( library.getId() );
+			
+			VersionedIdentifier libraryIdentifier = new VersionedIdentifier().withId(library.getName())
+					.withVersion(library.getVersion());
+			org.cqframework.cql.elm.execution.Library executable = libraryLoader.load(libraryIdentifier);
+			libraries.add(executable);
+	
+			for (RelatedArtifact related : library.getRelatedArtifact()) {
+				if (related.hasType() && related.getType().equals(RelatedArtifact.RelatedArtifactType.DEPENDSON)
+						&& related.hasResource()) {
+					org.hl7.fhir.r4.model.Library child = resolveLibrary(related.getResource(), libraryLoader,
+							libraryResourceProvider);
+					if (child != null) {
+						libraries.addAll(loadLibraries(child, libraryLoader, libraryResourceProvider, loadedIds));
+					}
+				}
+			}
+		}
+
+		return libraries;
+	}
+
+	/**
+	 * Resolve a FHIR Library resource based on the string reference provided.
+	 * 
+	 * @param resource                String reference to a FHIR Library resource
+	 * @param libraryLoader           LibraryLoader library loading mechanism for
+	 *                                handling CQL/ELM source to in memory library
+	 *                                loading
+	 * @param libraryResourceProvider Resource provider that handles loading the
+	 *                                FHIR library resource that contains the
+	 *                                CQL/ELM content
+	 * @return Loaded FHIR Library resource or null if the String does not contain
+	 *         an understood reference
+	 */
+	public static org.hl7.fhir.r4.model.Library resolveLibrary(String resource,
+			org.opencds.cqf.cql.engine.execution.LibraryLoader libraryLoader,
+			LibraryResolutionProvider<org.hl7.fhir.r4.model.Library> libraryResourceProvider) {
+
+		org.hl7.fhir.r4.model.Library library = null;
+
+		// Raw references to Library/libraryId or libraryId
+		if (resource.startsWith("Library/") || !resource.contains("/")) {
+			library = libraryResourceProvider.resolveLibraryById(resource.replace("Library/", ""));
+		}
+		// Full url (e.g. http://hl7.org/fhir/us/Library/FHIRHelpers)
+		else if (resource.contains(("/Library/"))) {
+			library = libraryResourceProvider.resolveLibraryByCanonicalUrl(resource);
+		}
```

Comment:
One thing I noticed in the [FHIR example library](http://hl7.org/fhir/us/cqfmeasures/Library-EXM125.json.html) is that the model info shows up as a library dependency.  I don't know enough about the "Library Model Info" dependency to make a statement on what to do here, but I figured I'd call it out as a future concern.

This might actually be related to the `TODO` you have right below here.

Reply:
This is enough for me to address. I will add the isLogicLibrary check back in.